backend/app/models/question.rb

```diff
@@ -1,7 +1,20 @@
 class Question < ApplicationRecord
   validates :name, presence: true
+  validate :validates_options, on: :update
   belongs_to :user
   belongs_to :question_type
   belongs_to :survey, optional: true
   has_many :options, dependent: :destroy
+
+  def single_choice?
+    question_type.name == QuestionType::SINGLE_CHOICE
```

Comment:
this method should be in the `QuestionType` class.
at this class you should use delegate to reference it

Reply:
Done!